src/crimson/osd/recovery_backend.cc

```diff
@@ -238,18 +238,17 @@ RecoveryBackend::scan_for_backfill(
 	if (pg.is_primary()) {
 	  obc = pg.obc_registry.maybe_get_cached_obc(object);
 	}
-	if (obc) {
-	  if (obc->obs.exists) {
```

Comment:
> scan_for_backill tries to load the metadata even if the obc is already in the cache while its existed is false

I might be missing something, when the above if case (`obc->obs.exists`) will be false and we will return in line 251, without loading the metadata. Right? 

Meaning we have 3 cases (before the patch):
1) obc found in cache && exists is true -> emplace it.
2) obc found in cache && exists is false -> return
3) obc not found -> load_metadata

Which case does the `obc && obc->obs.exists` comes to solve?

Reply:
Let me know if this is correct:
```

crimson/osd/recovery_backend: fix scan_for_backill metadata reload

Currently there are 3 cases (before the patch):

obc found in cache && exists is true -> emplace it.
obc found in cache && exists is false -> return
obc not found -> load_metadata

We should reload the metadata **also** when exists is false in order to
exclude the possibility that there's an ongoing obc load.

Fixes: https://tracker.ceph.com/issues/69154
Signed-off-by: Xuehan Xu <xuxuehan@qianxin.com>
```

Reply:
Yes, that's right:-) Will update the commit message.